src/evagram_input/input_tool/input_tool.py

```diff
@@ -15,41 +15,78 @@ def __init__(self, owner: str, experiment: str, eva_directory: str):
         self.eva_directory = eva_directory
         self.owner_id = None
         self.experiment_id = None
+
+        self._status = "OPEN"
+        self._diagnostics = 0
         self._conn = psycopg2.connect("host=127.0.0.1 port=5432 dbname=test_evagram user=postgres",
                                       password=os.getenv('DB_PASSWORD'))
         self._cursor = self._conn.cursor()
 
+    def __repr__(self):
+        message = f"Status: {self._status}\n"
+        message += f"Owner: {self.owner}\n"
+        message += f"Experiment: {self.experiment}\n"
+        message += f"Number of diagnostics added: {self._diagnostics}\n"
+        return message
+
     def input_data(self):
         try:
             self._verify_session_user()
             self.owner_id = self._add_current_user(self.owner)
             self.experiment_id = self._add_current_experiment(self.experiment, self.owner_id)
             self._run_task()
-        except psycopg2.OperationalError as err:
-            print(err)
+        except psycopg2.OperationalError:
+            self._status = "CLOSED AND ABORTED"
+            self._diagnostics = 0
+            raise
+        except FileNotFoundError:
+            self._status = "CLOSED AND ABORTED"
+            self._diagnostics = 0
+            raise
+        except RuntimeError:
+            self._status = "CLOSED AND ABORTED"
+            self._diagnostics = 0
+            raise
+        except RuntimeWarning:
+            self._status = "CLOSED WITH WARNING"
+            self._diagnostics = 0
+            raise
         else:
+            self._status = "CLOSED WITH SUCCESS"
             self._conn.commit()
         finally:
+            if self._status == "OPEN":
+                self._status = "CLOSED"
+            print("Terminating current workflow session...")
             self._cursor.close()
             self._conn.close()
+            print("Session object closed!")
+            print(self)
 
     def _run_task(self):
         eva_directory_path = self.eva_directory
+        files_found = 0
         for observation_dir in os.listdir(eva_directory_path):
             observation_path = os.path.join(eva_directory_path, observation_dir)
             if os.path.isdir(observation_path):
                 for plot in os.listdir(observation_path):
                     if plot.endswith(".pkl"):
                         self._add_plot(
                             eva_directory_path, observation_dir, plot, self.experiment_id)
+                        files_found += 1
+
+        if files_found == 0:
+            raise RuntimeWarning(("There was no diagnostics found in the given directory. "
+                                  "Make sure 'eva_directory' contains observation folders."))
```

Comment:
I decided to add an additional check to make sure diagnostics were found in the specified `eva_directory`. Since I thought this was more to inform the user that no diagnostics were added as a result of files not found, I raised a RuntimeWarning instead of an exception. Let me know what you think.

Reply:
Something like f'Make sure eva directory at {eva_directory_path} contains expected directory structure'. And then maybe we detail expected directory structure somewhere in evagram_input's documentation.

Reply:
Ok sounds good, should I change the error message to that instead?

Reply:
Yeah that would be good